### PR TITLE
(feat) OTA with `expo-updates`

### PR DESCRIPTION
**Pull request by mazenchami**
# Description

#182 

This PR adds in the necessary code for `expo-updates` and also runs through a couple tests to make sure its working as expected.

Additionally, we created a `Modal` component and upgraded to Expo48

# Graphics

[Loom Video](https://www.loom.com/share/0152a85f95f0435f991c99c1f4506f36)

# Checklist:

- [x] I have done a thorough self-review of my code
- [ ] I have tested with a screen reader and font-scaling turned on and added necessary accessibility features
- [x] I have run tests and linter

### CODE REVIEW

File: app/screens/DebugScreen.tsx [hunk 1]
```diff
@@ -44,18 +46,34 @@ export const DebugScreen: FC<StackScreenProps<AppStackParamList, "Debug">> = ()
...
+        <Text text={fcmToken} selectable />
+        <Button
+          shadowStyle={$resetStateButtonShadow}
+          tx="debugScreen.resetState"
+          onPress={() => clearState()}
+        />
+      </View>
+      <View style={$footer}>
+        <Text preset="companionHeading" text={"Specs:"} />
+        {Updates.channel.length > 0 ?? (
```
Comment (frankcalise) [inline app/screens/DebugScreen.tsx:65]: I wonder if the ?? operator is preventing this from displaying, maybe just a number is showing up somewhere for the length of the string? the others are ternary

File: eas.json [hunk 2]
```diff
@@ -44,6 +44,9 @@
       "env": {},
       "ios": {
         "resourceClass": "m1-medium"
+      },
+      "android": {
+        "buildType": "apk"
```
Comment (frankcalise) [inline eas.json:49]: probably want to revert this guy

File: app/app.tsx [hunk 3]
```diff
@@ -87,6 +92,53 @@ const CustomToast = () => {
   return <Toast config={toastConfig} topOffset={insets.top} />
 }
 
+// Setting up our OTA Updates component
+const OTAUpdates = () => {
+  const [isModalVisible, setIsModalVisible] = useState(false)
+
+  async function fetchAndRestartApp() {
+    await Updates.fetchUpdateAsync()
```
Comment (frankcalise) [inline app/app.tsx:100]: might want to still check this result `UpdateFetchResultSuccess` and `UpdateFetchResultFailure`

https://docs.expo.dev/versions/latest/sdk/updates/#updatefetchresult

Then if it fails can alert the user and just not reload.

Something to think about, if it does continuously fail they will have to press "Maybe later" quite a bit. Unless we want to add a don't remind me or something somewhere eventually (or if it failed) to async storage and skip future checks. IDK probably beyond the scope

File: app.config.ts [hunk 4]
```diff
@@ -90,5 +91,7 @@ export default ({ config }: ConfigContext): ExpoConfig => ({
     "@react-native-firebase/app",
     "@react-native-firebase/crashlytics",
     ["expo-build-properties", { ios: { useFrameworks: "static" } }],
+    ["expo-updates", { username: "infinitered" }],
```
Comment (frankcalise) [inline app.config.ts:94]: did we determine if we needed this AND the checkAutomatically flag? I mean it works so can leave it

**Review by frankcalise — COMMENTED** [review]
Left some notes, great work!

File: app.config.ts (hunk 4 above)
Comment (mazenchami) [reply app.config.ts:94]: yeah not 100% sure. don't want to mess with it since it works lol

File: app/screens/DebugScreen.tsx (hunk 1 above)
Comment (mazenchami) [reply app/screens/DebugScreen.tsx:65]: updating to `Updates.channel.length > 0 ? () : null` in next push

File: eas.json (hunk 2 above)
Comment (mazenchami) [reply eas.json:49]: reverting in next push

File: app/app.tsx (hunk 3 above)
Comment (mazenchami) [reply app/app.tsx]: fixed in next push

File: app/app.tsx [hunk 5]
```diff
@@ -87,6 +91,58 @@ const CustomToast = () => {
...
+    callback: onFetchUpdateAsync,
+  })
+
+  return (
+    <Modal
+      title="ota.title"
+      subtitle="ota.subtitle"
+      confirmOnPress={{
+        cta: () => {
+          fetchAndRestartApp()
```
Comment (frankcalise) [inline app/app.tsx:130]: do we need to await this? or just put a label somewhere with some progess like "updating..." ?

Comment (mazenchami) [reply app/app.tsx]: updated in next push: https://github.com/infinitered/ChainReactApp2023/pull/183/commits/2f3ab73f3a3ebd9cf43e62516ec90012aad73f0c

**Review by frankcalise — APPROVED** [review]
Looks good! :: passes to @robinheinze ::

**Comment by mazenchami** [comment]
Merging for now, will address comments @robinheinze & @morganick might have first thing on Monday!

File: app/app.tsx [hunk 6]
```diff
@@ -87,6 +91,61 @@ const CustomToast = () => {
   return <Toast config={toastConfig} topOffset={insets.top} />
 }
 
+// Setting up our OTA Updates component
+const OTAUpdates = () => {
```
Comment (robinheinze) [inline app/app.tsx:95]: Not critical. Would be nice to have this in it's own file to keep app.tsx more clean.

Comment (mazenchami) [reply app/app.tsx:95]: agreed. will be refactored in my other PR #189

**Pull request merged**